### ISSUE TITLE
dump_interactions giving error.

### ISSUE THREAD

**Issue by Shubham0209** [facebookresearch/EGG#255]
Actually, I have already trained my emergent game now I want to generate the interactions that are learned. Specofoacööy I want to pass my training dataset through my trained 'game' to store the messages that are generated and passed from sender to receiver. For this, I use the below command - 
`interaction_final = core.dump_interactions(game.to(DEVICE), train_loader, gs=True, variable_length=True,  device=DEVICE)`.

But I get the following error - 

![image](https://github.com/facebookresearch/EGG/assets/14998135/57486d07-dc5f-46b0-9690-3be2e464a752)

Any help, would be highly useful.

**Comment by robertodessi** [comment]
Hi @Shubham0209,

This doesn't look like the standard egg code, could you please add a bit more details on how you are calling these functions and what are dict1 and dict2?

Also, just to make sure, you trained your game, the one you passing to dump interactions, with gumbel softmax and length > 1, right?

**Comment by Shubham0209** [comment]
Hey @robertodessi , Thanks for your reply.

The above code snippet shows the code that is part of the interaction.py file [https://github.com/facebookresearch/EGG/blob/main/egg/core/interaction.py#LL59C1-L59C19](here). I am using the dump_interactions function from the same file to extract the interactions.

Yes, the game has been trained with the Gumbel softmax and length>1

**Comment by robertodessi** [comment]
Hi @Shubham0209,

I think I understood the problem. Could you please share the line(s) that are raising the error though? How's your code calling the `__add__` function or any code from the interaction file?

Thanks!

**Comment by Shubham0209** [comment]
@robertodessi here is the code:

`code_interactions = core.dump_interactions(game.to(DEVICE), dataset=train_loader, gs = True, variable_length = True, device = DEVICE
)`

**Comment by robertodessi** [comment]
Oh you wrote it already, sorry!

I think the error is here https://github.com/facebookresearch/EGG/blob/main/egg/core/interaction.py#L102

Could you maybe try to add these two checks at the beginning of the `_combine_aux_dicts` function?

```python
if not (dict1 or dict2):
    return {}

if bool(dict1) ^ bool(dict2):
    raise RuntimeError("Found an empty and non-empty dict when aggregating interactions")

```

**Comment by Shubham0209** [comment]
Yes, works now!

**Comment by robertodessi** [comment]
Great! Leaving it open as a reminder for myself to push the fix
